detect-plugins/cpu-microcode.py

```diff
@@ -0,0 +1,31 @@
+# ubuntu-drivers-common custom detect plugin for x86 CPU microcodes
+#
+# Author: Dimitri John Ledkov <dimitri.j.ledkov@intel.com>
+#
+# This plugin detects CPU microcode packages based on pattern matching
+# against the "vendor_id" line in /proc/cpuinfo.
+#
+# To add a new microcode family, simply insert a line into the db
+# variable with the following format:
+#
+# '<Pattern from your cpuinfo output>': '<Name of the driver package>',
+#
+
+import logging
+
+db = {
+    'GenuineIntel': 'intel-microcode',
+    'AuthenticAMD': 'amd64-microcode',
+     }
+
+def detect(apt_cache):
+    try:
+        with open('/proc/cpuinfo') as file:
+            for line in file:
+                if line.startswith('vendor_id'):
+                    cpu = line.split(':')[1].strip()
+                    return [db.get(cpu)]
```

Comment:
Yes that would be better. Also other plugins do not have return, or return None if nothing is detected. So that would work. Thanks for this. I'll land this now, to get into feature freeze =/ =)